toolkit/templates/forms/macros/forms.html

```diff
@@ -193,16 +197,16 @@
 {% macro upload(question_content, data, errors) -%}
   {%
     with
-    name=question_content.id,
-    question=question_content.question,
-    optional=question_content.optional,
-    question_advice=question_content.question_advice,
-    hint=question_content.hint,
-    value="Document uploaded {}".format(data[question_content.id]) if data[question_content.id] else None,
-    error=errors.get(question_content.id)['message'],
-    link=question_content.link,
+    error=kwargs.get("error", errors.get(question_content.id)['message']),
+    hint=kwargs.get("hint", question_content.hint),
+    link=kwargs.get("link", question_content.link),
+    message=kwargs.message,
+    name=kwargs.get("name", question_content.id),
+    optional=kwargs.get("optional", question_content.optional),
+    question=kwargs.get("question", question_content.question),
+    question_advice=kwargs.get("question_advice", question_content.question_advice),
     question_number=kwargs.question_number,
-    message=kwargs.message
+    value=kwargs.get("value", "Document uploaded {}".format(data[question_content.id]) if data[question_content.id] else None)
```

Comment:
Oooh this is weird.  But it's what was there already...

It's different in supplier frontend, so calls there will need to override the value as a kwarg when making the switch to use these.

Reply:
Yup & it does. https://github.com/alphagov/digitalmarketplace-supplier-frontend/pull/748/files#diff-712f48621dd493057d3678f370d01074L157